client/js/loading-error-handlers.js

```diff
@@ -0,0 +1,62 @@
+/* eslint strict: 0 */
+"use strict";
+
+/*
+ * This is a separate file for two reasons:
+ * 1. CSP policy does not allow inline javascript
+ * 2. It has to be a small javascript executed before all other scripts,
+ *    so that the timeout can be triggered while slow JS is loading
+ */
+
+(function() {
+	var displayReload = function displayReload() {
+		var loadingReload = document.getElementById("loading-reload");
+		if (loadingReload) {
+			loadingReload.style.display = "block";
+		}
+	};
+
+	var loadingSlowTimeout = setTimeout(function() {
+		var loadingSlow = document.getElementById("loading-slow");
+
+		if (loadingSlow) {
+			loadingSlow.style.display = "block";
+		}
+
+		displayReload();
```

Comment:
You can move this inside the `if` above, if there is no slow message, there won't be a reload button anyway.

Reply:
Well, I [moved the button outside the block](https://github.com/thelounge/lounge/pull/1845/files#diff-7f958a715ef0d644371ae8b1b5f18250R59) though.

I'm not 100% sure of why this `if` exists: is it because we're not taking any chances and it could not exist in case of a bug? If so then maybe we should display the button even if the text wasn't found, no?

Reply:
I didn't do this (at least yet) per my message above. Let me know what you think.

Reply:
The timer is not cancelled when the app loads, so it just does nothing if the element doesn't exist.

Reply:
Ah, good catch, will do and add a comment about this.

Reply:
Done, thanks!